hnn_core/parallel_backends.py

```diff
@@ -351,7 +353,7 @@ def requires_mpi4py(function):
         import mpi4py
         assert hasattr(mpi4py, '__version__')
         skip = False
-    except (ImportError, ModuleNotFoundError) as err:
```

Comment:
Same here, I don't see how this breaks PEP 8. I'm guessing you removed `ModuleNotFoundError` because it's redundant?

Reply:
Yeah the warning was B014, which catches redundant exception types. ModuleNotFoundError is a subclass of ImportError so excepting on ImportError would catch ModuleNotFoundError.